tests/utils.py

```diff
@@ -17,6 +18,13 @@
 TEST_DIR = Path(__file__).resolve().parent
 TEST_DATA_DIR = TEST_DIR / "data"
 
+# Pickle size tests often fail when run locally. At the moment, the only thing affecting the
+# pickled size is the path length. this is to estimate a safe offset for the size limit
+# based on the pickled sizes of the Path objects of the test data directories
+PICKLED_SIZE_OFFSET = len(pickle.dumps(TEST_DATA_DIR)) - len(
+    pickle.dumps(Path("home/runner/work/snap/snap/tests/data"))
+)
+
```

Comment:
Ah, this is since the pickled objects contain stuff like the the package and object name on top of the path.
E.g., in case of `Simulation` object,  the pickled object contains data like:
`bluepysnap.simulation`, `Simulation` 

So the pickled object looks something like
```
package_name
class_name
pathlib.Path  # The varying part
```
So what I do, is just to estimate the difference caused by the path as the size of the rest of it should remain somewhat constant.

Reply:
Yeah I see, but it seems cleaner to me to just define `PICKLED_SIZE_OFFSET = len(pickle.dumps(TEST_DATA_DIR))`, because `len(pickle.dumps(Path("home/runner/work/snap/snap/tests/data")))` is constant and equal to 88 and the path isn't special (I suppose it's the path used in the github CI).
Then it's possible to adjust the numbers used in tests (the upper limit is approximate, it doesn't need to be the exact number of bytes)
Maybe the variable can be called `PICKLED_SIZE_ADJUSTMENT` in this case.

However, I noticed that in some tests the path is pickled as a Path, while in others it's a string (because Circuit can be initialized with either of them, and no conversion is done):

- circuit.pkl -> path
- edge_population.pkl -> path
- edges.pkl -> str
- node_population.pkl -> str
- nodes.pkl -> str
- simulation.pkl -> path

The used space is different in the two cases, but an estimation based on the string length should be enough.
For consistency, we could also enforce a conversion from string to Path (or from Path to string) in `Circuit.__init__()`.

Reply:
Sure. TBH, `PICKLED_SIZE_OFFSET = len(pickle.dumps(TEST_DATA_DIR))` looks more elegant, anyway.

About the `Path`/`str` issue: I think we should just choose either and stick with that. I don't really see any benefit in storing it as a `Path` object so we might as well pickle the full path as a string. What are your thoughts?

Reply:
tbh in general I prefer to store paths as instances of Path, but since in this case the paths in Circuit and Simulation are saved only for pickling, I don't see any issue in storing them as strings.